src/program/services/filesystem/vfs/rivenvfs.py

```
@@ -1634,6 +1650,70 @@ async def open(self, inode: pyfuse3.InodeT, flags: int, ctx) -> pyfuse3.FileInfo
+                                    f"Found new file for {node.original_filename}"
+                                )
+
+                                break
+
+                            await trio.sleep(1)
+                except trio.TooSlowError:
+                    logger.error(
+                        f"Timeout waiting for new link to become available for path={node.path}"
+                    )
+
+                    raise pyfuse3.FUSEError(errno.ENOENT)
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)